scipy/linalg/_matfuncs_inv_ssq.py

```diff
@@ -298,15 +299,13 @@ def _logm_superdiag_entry(l1, l2, t12):
     """
     if l1 == l2:
         f12 = t12 / l1
-    elif abs(l1) < abs(l2) / 2 or abs(l2) < abs(l1) / 2:
+    elif abs(l2 - l1) > abs(l1 + l2) / 2:
         f12 = t12 * (np.log(l2) - np.log(l1)) / (l2 - l1)
     else:
         z = (l2 - l1) / (l2 + l1)
-        ua = _unwindk(np.log(l2) - np.log(l1))
-        ub = _unwindk(np.log(1+z) - np.log(1-z))
-        u = ua + ub
+        u = _unwindk(np.log(l2) - np.log(l1))
         if u:
-            f12 = t12 * (2*np.arctanh(z) + 2*np.pi*1j*(ua + ub)) / (l2 - l1)
+            f12 = t12 * 2 * (np.arctanh(z) + np.pi*1j*u) / (l2 - l1)
```

Comment:
I don't have access to the book, so I'll just assume you know what you're doing.